app.rb

```diff
@@ -0,0 +1,167 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'book'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+    @classroom = Classroom.new('Microverse')
+  end
+
+  def run
+    puts 'Welcome to School Library!'
+    loop do
+      menu
+      option = gets.chomp
+      break if option == '7'
+
+      get_num option
+    end
+    puts 'Thank you for using our Library!'
+  end
+
+  def menu
+    puts
+    puts 'Please choose an option by entering a number'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def get_num(option)
+    case option
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_by_person_id
+    else
+      puts 'Please enter a number between 1 and 7'
+    end
+  end
+
+  def list_all_books
+    puts 'No books are created yet! Please use option 5 to create a books.' if @books.empty?
+
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 0.75
+  end
+
+  def list_all_people
+    puts 'No people are created yet! Please use option 4 to create a person.' if @people.empty?
+    @people.map { |person| puts "Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    sleep 0.75
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or teacher (2) [Input respective number]: '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Please input 1 or 2'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+
+    student = Student.new(@classroom, age, name, parent_permission)
+    @people << student
+
+    puts 'Student successfully created!'
+    sleep 0.75
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
```

Comment:
- [ ] There seems to be a violation of the DRY and the Single-responsibility rules in the **`create_student`** and **`create_teacher`** methods. Both methods request for **`Age`** and **`Name`** inputs, this breaks the DRY rule. Also, user inputs and input validation should be handled by another in keeping with the Single-responsibility rule of SOLID. In keeping with both rules, kindly refactor your code accordingly.

  **EXAMPLE:** You can extract the requests into a new array and then return age and name in an array from the new method. Next, destructure the requests by calling the new method in **create_student** and **create_teacher**

  New method | Uses of new methods
  :---:|:---:
  ![lockless-person](https://user-images.githubusercontent.com/36542986/154949761-6a7cfe97-5c9d-4169-a79c-0bd1e6e7667c.jpg) | ![lockless-stud-teach](https://user-images.githubusercontent.com/36542986/154949764-1614bb83-ed0a-4100-af08-50a048934e84.jpg)